luigi/worker.py

```diff
@@ -123,7 +123,7 @@ def __init__(self, task, worker_id, result_queue, status_reporter,
         self.worker_id = worker_id
         self.result_queue = result_queue
         self.status_reporter = status_reporter
-        self.worker_timeout = task.worker_timeout or worker_timeout
+        self.worker_timeout = worker_timeout if task.worker_timeout is None else task.worker_timeout
```

Comment:
I find `task.worker_timeout if task.worker_timeout is not None else worker_timeout` might be easier to read, but no strong preference.